src/lib.rs

```diff
@@ -103,3 +127,40 @@ pub static GLOBAL: mycelium_alloc::Alloc = mycelium_alloc::Alloc;
 fn alloc_error(layout: core::alloc::Layout) -> ! {
     panic!("alloc error: {:?}", layout);
 }
+
+#[cfg(target_os = "none")]
+#[panic_handler]
+#[cold]
+fn panic(panic: &core::panic::PanicInfo) -> ! {
```

Comment:
i'm fine with moving this out of main, just curious if that was actually necessary or if you just did it because it made more sense?

Reply:
It's necessary because when the `[lib]` target is being built as a `harness = false` test target, it needs to have a panic handler defined, and the `main.rs` file isn't being included.

I tried making the `[[bin]]` target have the test (and turning off tests for the `[lib]` target), but that didn't work, as I couldn't `#[cfg(test)]` within `kernel_main`, as it was a separate crate.

Reply:
ah, that makes sense! i was just curious